File: system_api/cpuHandler.py
import cpuinfo
import psutil


def get_cpu_info():
    cpu_info = {
        "processor_name": cpuinfo.get_cpu_info()["brand_raw"],  # Processor Name
        "physical_cores": psutil.cpu_count(logical=False),  # Physical cores
        "total_cores": psutil.cpu_count(
            logical=True
        ),  # Total cores (physical + virtual)
        "core_usage": psutil.cpu_percent(percpu=True),  # CPU Usage Per Core
        "total_usage": psutil.cpu_percent(percpu=False),  # Total CPU Usage (all cores)
    }
    return cpu_info

File: system_api/cpu_handler.go
package main

import (
	"time"

	"github.com/shirou/gopsutil/cpu"
)

func GetCPUInfo() (map[string]interface{}, error) {
	cpuInfo := make(map[string]interface{})

	// Get CPU Info
	cpuInfoStat, err := cpu.Info()
	if err != nil {
		return nil, err
	}
	cpuInfo["processor_name"] = cpuInfoStat[0].ModelName

	// Get Physical Cores
	physicalCores, err := cpu.Counts(true)
	if err != nil {
		return nil, err
	}
	cpuInfo["physical_cores"] = physicalCores

	// Get Total Cores
	totalCores, err := cpu.Counts(false)
	if err != nil {
		return nil, err
	}
	cpuInfo["total_cores"] = totalCores

	// Get CPU Usage Per Core
	coreUsage, err := cpu.Percent(time.Duration(0), true)
	if err != nil {
		return nil, err
	}
	cpuInfo["core_usage"] = coreUsage

	// Get Total CPU Usage (all cores)
	totalUsage, err := cpu.Percent(time.Duration(0), false)
	if err != nil {
		return nil, err
	}
	cpuInfo["total_usage"] = totalUsage

	return cpuInfo, nil
}
